gpu/octree/include/pcl/gpu/octree/octree.hpp

```diff
@@ -144,8 +144,16 @@ namespace pcl
               * \param[in] queries array of centers
               * \param[out] result array of results ( one index for each query ) 
               */
+            PCL_DEPRECATED(1, 14, "use approxNearestSearch() which returns square distances instead")
```

Comment:
Couldn't we have both?
If one doesn't want the distances, you don't have to declare a container to save the distances in, which you will be forced to, if this one is removed?

Reply:
I don't have a strong opinion on this. On the one hand this function is more convenient for a user that doesn't want distances, and older code wouldn't need to be updated. On the other hand it's only a minor change with no significant performance hit and makes API more complex.

Reply:
I also don't have a strong opinion here. We can go with whatever you prefer.